docs/explanations/lifecycle.md

```diff
@@ -2,43 +2,44 @@
 
 The following demonstrates exactly what the code does with a plan through its lifecycle 
 of being written, loaded and run. Take the following plan.
-```
-    from typing import Any, List, Mapping, Optional, Union
-    
-    import bluesky.plans as bp
-    from blueapi.core import MsgGenerator
-    from dls_bluesky_core.core import inject
-    from bluesky.protocols import Readable
-    
-    
-    def count(
-        detectors: List[Readable] = [inject("det")],  # default valid for Blueapi only
-        num: int = 1,
-        delay: Optional[Union[float, List[float]]] = None,
-        metadata: Optional[Mapping[str, Any]] = None,
-    ) -> MsgGenerator:
-        """
-        Take `n` readings from a collection of detectors
-    
-        Args:
-            detectors (List[Readable]): Readable devices to read: when being run in Blueapi
-                                        defaults to fetching a device named "det" from its
-                                        context, else will require to be overridden.
-            num (int, optional): Number of readings to take. Defaults to 1.
-            delay (Optional[Union[float, List[float]]], optional): Delay between readings.
-                                                                Defaults to None.
-            metadata (Optional[Mapping[str, Any]], optional): Key-value metadata to include
-                                                            in exported data.
+
+```python
+from typing import Any, List, Mapping, Optional, Union
+
+import bluesky.plans as bp
+from blueapi.core import MsgGenerator
+from dls_bluesky_core.core import inject
+from bluesky.protocols import Readable
+
+
+def count(
+    detectors: List[Readable] = [inject("det")],  # default valid for Blueapi only
```

Comment:
Is this change going in prior to deprecating inject or after?

Reply:
Update to >=3.10 typing (List->list, Union[x, y] -> x | y, Optional[x] -> x | None)

Reply:
I am going to make a separate issue for moving all of the code snippets to Python files and checking that they still work and lint properly at docs build time, similar to ophyd-async.

Reply:
#677